src/api.js

```diff
@@ -42,9 +42,14 @@ export const request = (url, method, data, additionalHeaders = {}) => {
       .then(response => {
         if (response.status === 401) {
           const config = getConfig();
-          authenticate(config, true).then(t => {
-            resolve(authenticatedRequest(url, method, data, additionalHeaders));
-          });
+          // Prevent an endless loop which happens in case of re-authentication with the access token.
+          if (config.access_token === undefined) {
```

Comment:
Small/nit-picky.... 

It is generally consider safer practice to always use `typeof varName === "undefined"` over `varName === undefined`. [This link offers](https://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined) some good examples as to why it is good practice.